injector/disk_failure.go

```diff
@@ -102,8 +63,22 @@ func (i *DiskFailureInjector) Inject() error {
 	}
 
 	for _, path := range i.spec.Paths {
-		if err := i.config.Cmd.Run(pid, path, exitCode); err != nil {
-			return err
+		args := []string{"-p", strconv.Itoa(pid)}
+
+		if path != "" {
+			args = append(args, "-f", path)
+		}
+
+		if exitCode != 0 {
+			args = append(args, "-c", fmt.Sprintf("%v", exitCode))
+		}
+
+		cmd := i.config.CmdFactory.NewCmd(context.Background(), EBPFDiskFailureCmd, args)
+
+		bgCmd := command.NewBackgroundCmd(cmd, i.config.Log, i.config.ProcessManager)
```

Comment:
I don't find a way to mock this part 😢

Reply:
What I wanted to test for the CPU was:
* it does not matter how do you build your command, in the end I want to ensure "start" was called

https://github.com/DataDog/chaos-controller/blob/323596d971a465428ac8aa5180e7a77f900af0aa/injector/factory.go#L77

Hence I created a factory that will return a background command, factory that can be mocked for testing:
* https://github.com/DataDog/chaos-controller/blob/323596d971a465428ac8aa5180e7a77f900af0aa/injector/cpu_pressure_test.go#L43C23-L43C23

Background command hence can be a mock

If you really want to "mock" the creation of a command, you should provide a func as a argument that default to the `NewBackgroundCmd` as an example.

It really depends on how much you want to test and how.

Sometimes, doing only "component" testing is enough.

It's up to you to define at which granularity you need to go.